lua/bufferline/icons.lua

```diff
@@ -59,23 +59,20 @@ return {
       return '', ''
     end
 
-    local basename
-    local extension
-    local icon_char
-    local icon_hl
+    local basename, extension = '', ''
+    local filetype = buf_get_option(bufnr, 'filetype')
+    local icon_char, icon_hl = '', ''
 
     -- nvim-web-devicon only handles filetype icons, not other types (eg directory)
     -- thus we need to do some work here
     if filetype == 'netrw' or filetype == 'LuaTree' then
-      icon_char = ''
-      icon_hl = 'Directory'
+      icon_char, icon_hl = '', 'Directory'
     else
       if filetype == 'fugitive' or filetype == 'gitcommit' then
-        basename = 'git'
-        extension = 'git'
+        basename, extension = 'git', 'git'
       else
-        basename = fnamemodify(buffer_name, ':t')
-        extension = matchstr(basename, [[\v\.@<=\w+$]], '', '')
+        basename = fnamemodify(buf_get_name(bufnr), ':t')
+        extension = fnamemodify(basename, ':e')
```

Comment:
I'm using `fnamemodify` here, because I benchmarked it and found it to be ~10× faster.